File: Stepovy Vlad/StringCalculator.Library/StringCalculator.cs
namespace StringCalculator.Library
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text.RegularExpressions;

    public class StringCalculator
    {
        public int Add(string numbers)
        {
            if (numbers == string.Empty)
            {
                return 0;
            }
            var parsedNumbers = this.ParseString(numbers);

            var negativeNumbers = parsedNumbers.Where(number => number < 0).ToArray();
            if (negativeNumbers.Length > 0)
            {
                var negativeNumbersAsStrings = negativeNumbers.Select(number => number.ToString());
                throw new NegativesNotAllowedException(string.Join(",", negativeNumbersAsStrings));
            }

            return parsedNumbers.Where(number => number <= 1000).Sum();
        }

        private List<string> GetDelimeters(string numbers)
        {
            if (numbers.StartsWith("//"))
            {
                var delimetersSubstring = numbers.Substring(0, numbers.IndexOf('\n'));
                delimetersSubstring = delimetersSubstring.Substring(2);
                if (delimetersSubstring.Contains('[') &&
                    delimetersSubstring.Contains(']') &&
                    delimetersSubstring.Length > 1)
                {
                    Regex regex = new Regex(@"\[.*?\]");
                    var matches = regex.Matches(delimetersSubstring);
                    var delimeters = new List<string>();
                    foreach (Match match in matches)
                    {
                        delimeters.Add(match.Value);
                    }
                    for (int i = 0; i < delimeters.Count; i++)
                    {
                        delimeters[i] = delimeters[i].Substring(1, delimeters[i].Length - 2);
                    }
                    return delimeters;
                }
                else
                {
                    return new List<string> { delimetersSubstring };
                }
            }
            return new List<string>();
        }

        private int[] ParseString(string numbers)
        {
            var delimeters = this.GetDelimeters(numbers).ToList();
            delimeters.AddRange(new[] { ",", "\n" });
            if (numbers.StartsWith("//"))
            {
                numbers = numbers.Substring(numbers.IndexOf('\n'));
            }

            return numbers.Split(delimeters.ToArray(), StringSplitOptions.RemoveEmptyEntries)
                          .Select(number => Convert.ToInt32(number)).ToArray();
        }
    }
}

File: Snizhko Oleksandr/Lab1_Practice_Snizhko_Oleksandr/src/lab1_practice_snizhko_oleksandr/String_Calculator.java
package lab1_practice_snizhko_oleksandr;

import java.util.regex.Pattern;
import java.util.regex.Matcher;

/**
 *
 * @author Oleksandr
 */
public class String_Calculator {
    
    public static int adding(String text){ 
        
        if(text.isEmpty()){
            return 0; 
        } else if(text.contains(",")) {
            String[] numbers = text.split(",|\n");
            
            checkNegatives(numbers);
            
            int sum = 0;
            for (String number : numbers) {
                sum += Integer.parseInt(number);
            }
            return sum;
        } else if(text.startsWith("//")){
            Matcher m = Pattern.compile("//(.)\n(.*)").matcher(text);
            m.matches();  
            String delimeters = m.group(1);
            String numbers = m.group(2);
            
            String[] onlyNumbers = numbers.split(delimeters);
           
            checkNegatives(onlyNumbers);
            
            int sum = 0;
            for (String onlyNumber : onlyNumbers) {
                sum += Integer.parseInt(onlyNumber);
            }
            return sum;
        } else if(text.contains("\n")){
            String[] numbers = text.split("\n");
            
            checkNegatives(numbers);
            
            int sum = 0;
            for (String number : numbers) {
                sum += Integer.parseInt(number);
            }
            return sum;
        } else {
            return Integer.parseInt(text);
        }        
    }

    private static void checkNegatives(String[] values){
        String negatives = "";
        for(String value: values){
            if(Integer.parseInt(value)<0){
                negatives += ","+value;
            }
        }
        
        if(negatives.length() > 0){
            throw new RuntimeException(negatives.substring(1));
        }
    }
}